src/main/java/com/spikes2212/path/Path.java

```diff
@@ -1,94 +1,69 @@
 package com.spikes2212.path;
 
+import edu.wpi.first.wpilibj.Filesystem;
+
 import java.io.BufferedWriter;
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
-import java.util.Arrays;
-import java.util.LinkedList;
-import java.util.List;
+import java.nio.file.Paths;
+import java.util.*;
 
-/**
- * This class represents a path.
- *
- * @author T
- */
 public class Path {
 
     /**
-     * The points on the path.
-     */
-    private List<Waypoint> points;
-
-    /**
-     * Initializes a path.
-     * @param spacing the distance between two points filled in between points given as parameters
-     * @param smoothWeight how smooth to make the path, should be about 0.75 to 0.98
-     * @param tolerance the smoothing tolerance
-     * @param maxVelocity the robot's maximum velocity
+     * @param spacing         the distance between two points filled in between points given as parameters
+     * @param smoothWeight    how smooth to make the path, should be about 0.75 to 0.98
+     * @param tolerance       the smoothing tolerance
+     * @param maxVelocity     the robot's maximum velocity
      * @param turningConstant speed constant at curves (the higher it is, the faster you turn)
      * @param maxAcceleration the robot's maximum acceleration
-     * @param points the initial points on the path. Apart from the edges, non of the points are guaranteed
-     *               to be on the final path
+     * @param points          the initial points on the path. Apart from the edges, non of the points are guaranteed
+     *                        to be on the final path
+     * @return the generated path
      */
-    public Path(double spacing
-            , double smoothWeight, double tolerance,
-                double maxVelocity, double turningConstant
-            , double maxAcceleration, Waypoint... points) {
-        this.points = new LinkedList<>(Arrays.asList(points));
-        generate(spacing, smoothWeight, tolerance
-                , maxVelocity, turningConstant, maxAcceleration);
-    }
-
-    private Path(List<Waypoint> points) {
-        this.points = points;
-    }
-
-    public List<Waypoint> getPoints() {
+    public static List<Waypoint> generate(List<Waypoint> points, double spacing, double smoothWeight, double tolerance,
+                                          double maxVelocity, double turningConstant, double maxAcceleration) {
+        fill(points, spacing);
```

Comment:
it is better to make all those methods return a new list rather than mutating an existing one, because some lists might be immutable (such as the those created using `Arrays.asList`)